src/java.base/share/classes/java/util/zip/ZipFile.java

```diff
@@ -1222,16 +1222,17 @@ private int checkAndAddEntry(int pos, int index)
             int nlen = CENNAM(cen, pos);
             int elen = CENEXT(cen, pos);
             int clen = CENCOM(cen, pos);
-            if (entryPos + nlen > cen.length - ENDHDR) {
+            long headerSize = (long)CENHDR + nlen + clen + elen;
```

Comment:
I can remove the cast as that was a holdover.  I chose to  make this a long knowing that it would not overflow but  an overflow while unlikely could occur depending on the value of `pos`   in the statement below

```
if (headerSize > 0xFFFF || pos + headerSize > cen.length - ENDHDR) {
                zerror("invalid CEN header (bad header size)");
 }
```

I could keep headerSize an int and then move the cast to the if statement:

```
if (headerSize > 0xFFFF || (long)pos + headerSize > cen.length - ENDHDR) {
                zerror("invalid CEN header (bad header size)");
 }
```
I  decided making headerSize a long  might be clearer but do not have a strong preference and will go with the consensus 

As far as the name,  I don't have a strong preference, but not sure combinedLength is any better